lib/autoInject.js

```diff
@@ -108,8 +117,6 @@ export default function autoInject(tasks, callback) {
             taskFn = params.pop();
 
             newTasks[key] = params.concat(params.length > 0 ? newTask : taskFn);
-        } else if (taskFn.length === 0) {
-            throw new Error("autoInject task functions require explicit parameters.");
```

Comment:
I dont like this side effect.  It will catch a really common pitfall where people pass bound (or otherwise wrapped) functions to `autoInject`.  Could you change the conditional to be more like `taskFn.length === 0 && hasNoDefaultParams`?